### PR TITLE
Update the upgrade instructions

### PR DESCRIPTION
**Pull request by weshayutin**

### CODE REVIEW

File: docs/upgrade_1-2_to_1-3.md
Comment (mateusoliveira43) [inline docs/upgrade_1-2_to_1-3.md:1]: Please address these comments:
- https://github.com/openshift/oadp-operator/pull/1171#discussion_r1376500659
- https://github.com/openshift/oadp-operator/pull/1171#discussion_r1376501450

File: docs/upgrade_1-1_to_1-2.md [hunk 1]
```diff
@@ -19,10 +19,21 @@
...
+
+### Upgrade the OADP Operator
+
+For general operator upgrade instructions please review the [OpenShift documentation](https://docs.openshift.com/container-platform/4.13/operators/admin/olm-upgrading-operators.html)
+* Change the Subscription for the OADP Operator from `stable-1.1` to `stable-1.2`
+* Allow time for the operator and containers to update and restart
+
 ### Convert your DPA to the new version
 
 If you are using fields that were updated in `spec.configuration.velero.args`, you need to update there new names. Example
```
Comment (mateusoliveira43) [inline docs/upgrade_1-1_to_1-2.md:39]: I would cite `Backup the DPA configuration` section here

File: docs/upgrade_1-2_to_1-3.md [hunk 2]
```diff
@@ -36,21 +36,44 @@
...
+
+### Backup the DPA configuration
 
 Save your current DataProtectionApplication (DPA) CustomResource config, be sure to remember the values.
 
+For example:
+```
+oc get dpa -n openshift-adp -o yaml > dpa.orig.backup 
+```
+
```
Comment (mateusoliveira43) [inline docs/upgrade_1-2_to_1-3.md:64]: add section `### Upgrade the OADP Operator`

File: docs/upgrade_1-2_to_1-3.md [hunk 3]
```diff
@@ -36,21 +36,44 @@
 +      uploaderType: restic
 ```
 
-> **Note:** OADP will be favoring Kopia over Restic in the near future.
+> **Note:** In the next version of OADP the Restic option will be deprecated and Kopia will be come the default uploaderType.
 
 ## Upgrade steps
 
-### Create new backup
+### If the OADP 1.2 tech-preview DataMover feature is in use, please read the following.
```
Comment (mateusoliveira43) [inline docs/upgrade_1-2_to_1-3.md:43]: NIT: no final dot in section title

File: docs/upgrade_1-2_to_1-3.md [hunk 4]
```diff
@@ -36,21 +36,44 @@
...
 
-If you are using DataMover, you need to create new backups before upgrade, because the 1.2 DataMover backups will not work after upgrade. We suggest to create Restic backups.
+OADP 1.2 DataMover backups can NOT be restored with OADP 1.3. To prevent a gap in the data protection of your applications we recommend the following.
 
-### Copy old DPA
+* If on cluster backups are sufficient and CSI storage is available
+  * Backup the applications with a CSI backup
+
+* If off cluster backups are required
+  * Backup the applications with a filesystem backup using the `--default-volumes-to-fs-backup=true` option.
```
Comment (mateusoliveira43) [inline docs/upgrade_1-2_to_1-3.md:51]: Since this will be in 1.2, we can say its restic, right?

would also tell about CLI x UI

"Backup the applications with restic filesystem backup using the `--default-volumes-to-fs-backup=true` flag, if using velero CLI, or using `defaultVolumesToFsBackup` field in Backup CR."

File: docs/upgrade_1-2_to_1-3.md [hunk 5]
```diff
@@ -36,21 +36,44 @@
...
+```
+
+For general operator upgrade instructions please review the [OpenShift documentation](https://docs.openshift.com/container-platform/4.13/operators/admin/olm-upgrading-operators.html)
+* Change the Subscription for the OADP Operator from `stable-1.2` to `stable-1.3`
+* Allow time for the operator and containers to update and restart
+
 ### Convert your DPA to the new version
 
-If you are using DataMover, you need to update with the new configuration. Example
+If you are using DataMover, you need to update with the new configuration.
```
Comment (mateusoliveira43) [inline docs/upgrade_1-2_to_1-3.md:71]: To address https://github.com/openshift/oadp-operator/pull/1171#discussion_r1376505384 we could change to

"If you were using DataMover in 1.2, you need to update your DPA with the new configuration, after upgrading to 1.3."

File: docs/upgrade_1-2_to_1-3.md [hunk 6]
```diff
@@ -36,21 +36,44 @@
...
+* Allow time for the operator and containers to update and restart
+
 ### Convert your DPA to the new version
 
-If you are using DataMover, you need to update with the new configuration. Example
+If you are using DataMover, you need to update with the new configuration. 
+
+* remove the features.dataMover key and values from DPA
+* remove the VSM plugin 
+
```
Comment (mateusoliveira43) [inline docs/upgrade_1-2_to_1-3.md:75]: Also add something like

"Also, if you want to use 1.3 new DataMover, along with CSI plugin, you need to enable kopia."

?

File: docs/upgrade_1-1_to_1-2.md (hunk 1 above)
Comment (weshayutin) [reply docs/upgrade_1-1_to_1-2.md:40]: Upgrading the operator is not really tied to the DPA.   I would to prefer to leave it as is.  We'll go through it together and see if it makes sense later.

File: docs/upgrade_1-2_to_1-3.md (hunk 2 above)
Comment (weshayutin) [reply docs/upgrade_1-2_to_1-3.md:66]: I don't understand this comment

File: docs/upgrade_1-2_to_1-3.md (hunk 3 above)
Comment (weshayutin) [reply docs/upgrade_1-2_to_1-3.md]: good call

File: docs/upgrade_1-2_to_1-3.md [hunk 7]
```diff
@@ -36,21 +36,44 @@
...
-If you are using DataMover, you need to create new backups before upgrade, because the 1.2 DataMover backups will not work after upgrade. We suggest to create Restic backups.
+OADP 1.2 DataMover backups can NOT be restored with OADP 1.3. To prevent a gap in the data protection of your applications we recommend the following.
 
-### Copy old DPA
+* If on cluster backups are sufficient and CSI storage is available
+  * Backup the applications with a CSI backup
+
+* If off cluster backups are required
+  * Backup the applications with a filesystem backup using the `--default-volumes-to-fs-backup=true` option.
+  * Backup the applications with your CloudStorage plugins e.g. velero-plugin-for-aws
```
[the remark this answers is not in the document]
Comment (weshayutin) [reply docs/upgrade_1-2_to_1-3.md:54]: No.. csi can utilize the on cluster storage.

File: docs/upgrade_1-2_to_1-3.md (hunk 4 above)
Comment (weshayutin) [reply docs/upgrade_1-2_to_1-3.md:53]: agree.. thanks

File: docs/upgrade_1-2_to_1-3.md (hunk 6 above)
Comment (weshayutin) [reply docs/upgrade_1-2_to_1-3.md:77]: You do not need the csi plugin w/ kopia nor do you need kopia to use the csi plugin.

File: docs/upgrade_1-2_to_1-3.md [hunk 8]
```diff
@@ -42,15 +42,15 @@
...
 
 OADP 1.2 Data Mover backups can **NOT** be restored with OADP 1.3. To prevent a gap in the data protection of your applications we recommend the following to be **completed prior to the OADP upgrade**.
 
 * If on cluster backups are sufficient and CSI storage is available
   * Backup the applications with a CSI backup
 
 * If off cluster backups are required
-  * Backup the applications with a filesystem backup using the `--default-volumes-to-fs-backup=true` option.
+  * Backup the applications with a filesystem backup using the `--default-volumes-to-fs-backup=true` or `backup.spec.defaultVolumesToFsBackup` options.
   * Backup the applications with your CloudStorage plugins e.g. velero-plugin-for-aws
```
Comment (kaovilai) [inline docs/upgrade_1-2_to_1-3.md:54]: I would rename this object storage plugins for clarity and consistency with velero lingo.

In addition, we have CloudStorage API that can be confused with this wording.

File: docs/upgrade_1-2_to_1-3.md [hunk 9]
```diff
@@ -1,21 +1,23 @@
...
+    From this update, OADP 1.3 now uses the Velero Built-in Data Mover instead of the VSM/Volsync DataMover. This changes the following:
 
-    - `spec.features.dataMover` (you can delete secret for `spec.features.dataMover.credentialName` from your cluster, if you want) and `vsm` plugin are not necessary anymore and you need to remove them from your DPA
+    - The `spec.features.dataMover` field and the `vsm` plugin are not compatible with 1.3 and must be removed from the DPA configuration.
 
-    - `volsync` operator is not necessary anymore (you can uninstall it from your cluster, if you want)
+    - The `spec.features.dataMover.credentialName`field can safely be removed
 
-    - `volumesnapshotbackups.datamover.oadp.openshift.io` and `volumesnapshotrestores.datamover.oadp.openshift.io` CustomResourceDefinitions are not necessary anymore (you can delete `volumesnapshotbackups.datamover.oadp.openshift.io` and `volumesnapshotrestores.datamover.oadp.openshift.io` CRDs from your cluster, if you want)
+    - The Volsync operator is no longer required and can optionally be removed.
```
Comment (PrasadJoshi12) [inline docs/upgrade_1-2_to_1-3.md:14]: Additionally we can remove the datamover secret and volsync privileged annotation from the namespace.

File: docs/upgrade_1-1_to_1-2.md [hunk 10]
```diff
@@ -5,24 +5,36 @@
...
+    - The `default-volumes-to-restic` field was renamed `default-volumes-to-fs-backup`, **if you are using `spec.velero`, you need to add it back, with the new name, to your DPA after upgrading OADP**
 
-    - `default-restic-prune-frequency` was renamed `default-repo-maintain-frequency`, **if you are using it, you need to add it back, with the new name, to your DPA after upgrading OADP**
+    - The `default-restic-prune-frequency` field was renamed `default-repo-maintain-frequency`, **if you are using it, you need to add it back, with the new name, to your DPA after upgrading OADP**
 
-    - `restic-timeout` was renamed `fs-backup-timeout`, **if you are using it, you need to add it back, with the new name, to your DPA after upgrading OADP**
+    - The `restic-timeout` field was renamed `fs-backup-timeout`, **if you are using it, you need to add it back, with the new name, to your DPA after upgrading OADP**
 
-- `restic` DaemonSet was renamed to `node-agent` (no changes required, OADP code handles this change)
+- The `restic` DaemonSet was renamed to `node-agent`.  OADP will automatically update this field)
```
Comment (mateusoliveira43) [inline docs/upgrade_1-1_to_1-2.md:16]: Would undo this change, DaemonSet is not a field

File: docs/upgrade_1-1_to_1-2.md [hunk 11]
```diff
@@ -5,24 +5,36 @@
 
 - Velero was updated from version 1.9 to 1.11 (Changes reference: https://velero.io/docs/v1.11/upgrade-to-1.11/#upgrade-from-version-lower-than-v1100)
 
-    From this update, in `spec.configuration.velero.args` these were changed:
+    From this update, in the DPA's configuration `spec.configuration.velero.args` have changed:
 
-    - `default-volumes-to-restic` was renamed `default-volumes-to-fs-backup`, **if you are using it, you need to add it back, with the new name, to your DPA after upgrading OADP**
+    - The `default-volumes-to-restic` field was renamed `default-volumes-to-fs-backup`, **if you are using `spec.velero`, you need to add it back, with the new name, to your DPA after upgrading OADP**
```
Comment (mateusoliveira43) [inline docs/upgrade_1-1_to_1-2.md:10]: would revert `spec.velero` to it

File: docs/upgrade_1-2_to_1-3.md [hunk 12]
```diff
@@ -1,21 +1,23 @@
 # Upgrading from OADP 1.2
 
-> **NOTE:** Always upgrade to next minor version, do NOT skip versions. To update to higher version, do one upgrade at a time. Example: to upgrade from 1.1 to 1.3, upgrade first to 1.2, then to 1.3.
+> **NOTE:** Always upgrade to next minor version, do NOT skip versions. To update to higher version, please upgrade one channel at a time. Example: to upgrade from 1.1 to 1.3, upgrade first to 1.2, then to 1.3.
```
Comment (mateusoliveira43) [inline docs/upgrade_1-2_to_1-3.md:3]: also change this in the other file

File: docs/upgrade_1-2_to_1-3.md (hunk 2 above)
Comment (mateusoliveira43) [reply docs/upgrade_1-2_to_1-3.md:66]: to have the same structure as the other file

File: docs/upgrade_1-1_to_1-2.md [hunk 13]
```diff
@@ -5,24 +5,36 @@
...
 
-    - `restic-timeout` was renamed `fs-backup-timeout`, **if you are using it, you need to add it back, with the new name, to your DPA after upgrading OADP**
+    - The `restic-timeout` field was renamed `fs-backup-timeout`, **if you are using it, you need to add it back, with the new name, to your DPA after upgrading OADP**
 
-- `restic` DaemonSet was renamed to `node-agent` (no changes required, OADP code handles this change)
+- The `restic` DaemonSet was renamed to `node-agent`.  OADP will automatically update this field)
 
-- `resticrepositories.velero.io` CustomResourceDefinition was renamed to `backuprepositories.velero.io` (you can delete `resticrepositories.velero.io` CRD from your cluster, if you want)
+- The CustomResourceDefinition `resticrepositories.velero.io` was renamed to `backuprepositories.velero.io` 
+  * The CustomResourceDefinition `resticrepositories.velero.io` can optionally be removed from the cluster.
```
Comment (mateusoliveira43) [inline docs/upgrade_1-1_to_1-2.md:19]: to follow structure from the other file, would change to "The CustomResourceDefinition `resticrepositories.velero.io` is no longer required and can optionally be removed."

File: docs/upgrade_1-2_to_1-3.md [hunk 14]
```diff
@@ -1,21 +1,23 @@
...
 ## Changes from OADP 1.2 to 1.3
 
-- Velero was updated from version 1.11 to 1.12 (Changes reference: https://velero.io/docs/v1.12/upgrade-to-1.12/#upgrade-from-v110-or-higher)
+- The Velero server has been updated from version 1.11 to 1.12 (Changes reference: https://velero.io/docs/v1.12/upgrade-to-1.12/#upgrade-from-v110-or-higher)
 
-    From this update, OADP now uses Velero Built-in DataMover instead of VSM/Volsync DataMover. This change the following:
+    From this update, OADP 1.3 now uses the Velero Built-in Data Mover instead of the VSM/Volsync DataMover. This changes the following:
```
Comment (mateusoliveira43) [inline docs/upgrade_1-2_to_1-3.md:8]: there is both `DataMover` and `Data Mover` in the file. Would use just one wording in whole file

File: docs/upgrade_1-2_to_1-3.md [hunk 15]
```diff
@@ -1,21 +1,23 @@
...
 
-    - `spec.features.dataMover` (you can delete secret for `spec.features.dataMover.credentialName` from your cluster, if you want) and `vsm` plugin are not necessary anymore and you need to remove them from your DPA
+    - The `spec.features.dataMover` field and the `vsm` plugin are not compatible with 1.3 and must be removed from the DPA configuration.
 
-    - `volsync` operator is not necessary anymore (you can uninstall it from your cluster, if you want)
+    - The `spec.features.dataMover.credentialName`field can safely be removed
```
Comment (mateusoliveira43) [inline docs/upgrade_1-2_to_1-3.md:12]: its not the field, its the secret user previously created

File: docs/upgrade_1-2_to_1-3.md (hunk 9 above)
Comment (mateusoliveira43) [reply docs/upgrade_1-2_to_1-3.md:14]: how is this annotation @PrasadJoshi12 ? I never manually added any when using 1.2 datamover

File: docs/upgrade_1-2_to_1-3.md (hunk 8 above)
Comment (weshayutin) [reply docs/upgrade_1-2_to_1-3.md:54]: good point

File: docs/upgrade_1-2_to_1-3.md (hunk 9 above)
Comment (weshayutin) [reply docs/upgrade_1-2_to_1-3.md:14]: good point.. addressing it now.

File: docs/upgrade_1-1_to_1-2.md (hunk 10 above)
Comment (weshayutin) [reply docs/upgrade_1-1_to_1-2.md:16]: ur right.. changed to `OADP will automatically update the DaemonSet name.`

File: docs/upgrade_1-2_to_1-3.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (weshayutin) [reply docs/upgrade_1-2_to_1-3.md:54]: done

File: docs/upgrade_1-2_to_1-3.md (hunk 2 above)
Comment (mateusoliveira43) [reply docs/upgrade_1-2_to_1-3.md:68]: resolved

File: docs/upgrade_1-2_to_1-3.md (hunk 14 above)
Comment (mateusoliveira43) [reply docs/upgrade_1-2_to_1-3.md]: resolved

File: docs/upgrade_1-2_to_1-3.md (hunk 15 above)
Comment (mateusoliveira43) [reply docs/upgrade_1-2_to_1-3.md]: resolved

File: docs/upgrade_1-2_to_1-3.md [hunk 16]
```diff
@@ -1,21 +1,25 @@
...
+    - The Volsync operator is no longer required and can optionally be removed.
 
-    - `volumesnapshotbackups.datamover.oadp.openshift.io` and `volumesnapshotrestores.datamover.oadp.openshift.io` CustomResourceDefinitions are not necessary anymore (you can delete `volumesnapshotbackups.datamover.oadp.openshift.io` and `volumesnapshotrestores.datamover.oadp.openshift.io` CRDs from your cluster, if you want)
+    - The CustomResourceDefinitions `volumesnapshotbackups.datamover.oadp.openshift.io` and `volumesnapshotrestores.datamover.oadp.openshift.io` are no longer required and can optionally be removed.
 
-    Also, OADP now supports a new file system backup software: Kopia.
+    - The secrets used for the OADP-1.2 Data Mover are no longer required and can optionally be removed.
 
-    - To use it, use the new `spec.configuration.nodeAgent` field. Example
+    - The `velero.io/csi-volumesnapshot-class` label in the VolumeSnapShotClass manifests are no longer required and can optionally be removed.
```
Comment (mateusoliveira43) [inline docs/upgrade_1-2_to_1-3.md:18]: we still need this. Ref: https://velero.io/docs/v1.12/csi/ Thanks @kaovilai

**Review by mateusoliveira43 — CHANGES_REQUESTED** [review]
address `velero.io/csi-volumesnapshot-class` comment

**Review by mateusoliveira43 — APPROVED** [review]
